sdk/core/core-http/src/policies/proxyPolicy.ts

```diff
@@ -100,11 +103,14 @@ export function getDefaultProxySettings(proxyUrl?: string): ProxySettings | unde
 }
 
 export function proxyPolicy(proxySettings?: ProxySettings): RequestPolicyFactory {
-  if (!proxySettings) {
-    proxySettings = getDefaultProxySettings();
-  }
   return {
     create: (nextPolicy: RequestPolicy, options: RequestPolicyOptions) => {
+      if (!proxySettings) {
+        proxySettings = getDefaultProxySettings();
```

Comment:
assuming that `proxySettings` aren't passed to `proxyPolicy`, this will now re-fetch the defaults every time a request is made, since `core-http` calls `create` on each request, which isn't what I think you want?

Reply:
You are right! It should be called when creating the policy factory.

Reply:
If these environment variables are changed after we create the pipeline then we won't know, but it is still better than current behavior.

Reply:
One problem with current approach is that we load the noProxyList from NO_PROXY env var and cache it in first `proxyPolicy()` thus
- it is not possible to change the list when creating new pipelines.
- it is not easy to configure noProxyList programmatically. For example, storage explorer allow users to configure proxy settings from the UI instead of using the NO_PROXY environment variable.

To address 1) we should remove the caching and load the list every time `proxyPolicy()` is called.

For 2) maybe add a parameter for noProxyList?

```ts
export function proxyPolicy(proxySettings?: ProxySettings, customNoProxyList?: string[]): RequestPolicyFactory {
  if (!proxySettings) {
    proxySettings = getDefaultProxySettings();
  }
  if (customNoProxyList) {
    noProxyList = customNoProxyList;
  } else {
    noProxyList = loadNoProxy();
  }

  return {
    create: (nextPolicy: RequestPolicy, options: RequestPolicyOptions) => {
      return new ProxyPolicy(nextPolicy, options, proxySettings!);
    }
  };
}
```
Another question is do we want to make noProxyList per ProxyPolicy, similar to proxySettings?  There might be case where two clients from different SDK libraries want to use different set of proxy configurations. This will then make the noProxyList and the byPassedList cache per ProxyPolicy 

@bterlson @chradek @xirzec

Reply:
logged #15133 for further discussion